src/components/sidebar/sidebar.test.tsx

```diff
@@ -0,0 +1,318 @@
+    await user.keyboard("{a}");
+
+    expect(screen.getByRole("dialog")).toBeVisible();
+    expect(onCancel).not.toHaveBeenCalled();
+  });
+
+  test("focus returns to last focused element when sidebar is closed", async () => {
+    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+    render(
+      <CarbonProvider>
+        <MockApp onCancel={() => {}} />
+      </CarbonProvider>
+    );
+
+    await user.click(screen.getByRole("button", { name: "Close" }));
+
+    await waitForElementToBeRemoved(() => screen.queryByRole("dialog"));
+    expect(document.body).toHaveFocus();
+  });
+});
+
+test("focus is trapped within sidebar when opened", async () => {
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+  render(
+    <CarbonProvider>
+      <Sidebar open header="My sidebar">
+        <button type="button">First</button>
+        <button type="button">Second</button>
+      </Sidebar>
+    </CarbonProvider>
+  );
+  const firstButton = screen.getByRole("button", { name: "First" });
+  firstButton.focus();
+
+  await user.tab();
+
+  expect(screen.getByRole("button", { name: "Second" })).toHaveFocus();
+
+  await user.tab();
+
+  expect(screen.getByRole("button", { name: "First" })).toHaveFocus();
+});
+
+test("focus is not trapped within sidebar when enableBackgroundUI is true", async () => {
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+  render(
+    <CarbonProvider>
+      <Sidebar open header="My sidebar" enableBackgroundUI>
+        <button type="button">First</button>
+        <button type="button">Second</button>
+      </Sidebar>
+    </CarbonProvider>
+  );
+  const firstButton = screen.getByRole("button", { name: "First" });
+  firstButton.focus();
+
+  await user.tab();
+
+  expect(screen.getByRole("button", { name: "Second" })).toHaveFocus();
+
+  await user.tab();
+
+  expect(firstButton).not.toHaveFocus();
+});
+
+test("can refocus sidebar container using a forwarded ref", async () => {
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+  const MockApp = () => {
+    const ref = React.useRef<HTMLDivElement | null>(null);
+    return (
+      <Sidebar open header="My sidebar" ref={ref}>
+        <button type="button" onClick={() => ref.current?.focus()}>
+          Refocus sidebar
+        </button>
+      </Sidebar>
+    );
+  };
+  render(
+    <CarbonProvider>
+      <MockApp />
+    </CarbonProvider>
+  );
+
+  await user.click(screen.getByRole("button", { name: "Refocus sidebar" }));
+
+  await waitFor(() => {
+    expect(screen.getByRole("dialog")).toHaveFocus();
+  });
+});
+
+test("calls a forwarded ref callback on render with sidebar element as argument", () => {
+  const ref = jest.fn();
+  render(<Sidebar open header="My sidebar" ref={ref} />);
+
+  expect(ref).toHaveBeenNthCalledWith(1, screen.getByRole("dialog"));
+});
+
+test("ensures overflowing content is scrollable", () => {
```

Comment:
Question (non-blocking): is this needed for coverage, this feels like something we can capture in chromatic?

Reply:
Yeah, that's right. It's mainly because we have some logic to delegate the responsibility of handling overflow to a child `Form`.